great_expectations/execution_engine/execution_engine.py

```diff
@@ -483,19 +487,16 @@ def add_column_row_condition(
                 "Base ExecutionEngine does not support adding nan condition filters"
             )
 
-        if "row_condition" in domain_kwargs and domain_kwargs["row_condition"]:
-            raise ge_exceptions.GreatExpectationsError(
-                "ExecutionEngine does not support updating existing row_conditions."
-            )
-
         new_domain_kwargs = copy.deepcopy(domain_kwargs)
         assert "column" in domain_kwargs or column_name is not None
         if column_name is not None:
             column = column_name
         else:
             column = domain_kwargs["column"]
-        new_domain_kwargs["condition_parser"] = "great_expectations__experimental__"
-        new_domain_kwargs["row_condition"] = f'col("{column}").notnull()'
+        row_condition: RowCondition = RowCondition(
+            condition=f'col("{column}").notnull()', type_=RowConditionParserType.GE
+        )
+        new_domain_kwargs.setdefault("filter_conditions", []).append(row_condition)
```

Comment:
I can't believe I didn't know about this method 👀

Reply:
I just found it for this PR - super useful!